components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -789,6 +793,24 @@ setup_file (      IOFileSpecs& filespecs,
   auto mode = m_resume_output_file ? scorpio::Append : scorpio::Write;
   scorpio::register_file(filename,mode,filespecs.iotype);
   if (m_resume_output_file) {
+    // We may have resumed an output file that contains extra snapshots *after* the restart time.
+    // E.g., if we output every step and the run crashed a few steps after writing the restart.
+    // In that case, we need to reset the time dimension in the output file, so that the extra
+    // snapshots will be overwritten.
+    const auto all_times = scorpio::get_all_times(filename);
+    int ntimes = all_times.size();
+    int ngood  = 0;
+    for (const auto& t : all_times) {
+      auto keep = t<=m_output_control.last_write_ts.days_from(m_case_t0);
+      if (keep) {
+        ++ngood;
+      } else {
+        break;
+      }
+    }
+    if (ngood<ntimes) {
+      scorpio::reset_unlimited_dim_len(filename,ngood);
```

Comment:
This resets stuff on the EAMxx data structures side, not in scorpio. When we open an existing file, we read the current time dim len. When we write new snaps for vars, we write them at position `time_dim.len + 1`, and we call `update_time` we basically do `++time_dim.len`. So, without this we would append new snapshots, rather than write them at the proper location.

Reply:
To reiterate: changing `time_dim.length` does not change the length of the time dim in scorpio. If you were to reset the time dim length from 10 to 1, write one snap and exit, the file would still contain 10 snapshots; you would have simply overwritten the 2nd. I did not spend any effort to "remove" the slices from the file (I don't even know if it's possible); it's pointless, since we will overwrite them anyways before we close it.

This routine simply allows EAMxx to reset the time index at which it will start writing snapshots.